File: visual-model-ui/src/main/java/org/visual/model/ui/bar/WindowsTitleBar.java
package org.visual.model.ui.bar;


import lombok.val;
import org.kordamp.ikonli.fluentui.FluentUiFilledAL;
import org.kordamp.ikonli.fluentui.FluentUiFilledMZ;
import org.kordamp.ikonli.javafx.FontIcon;
import org.visual.model.ui.TitleBar;
import org.visual.model.ui.control.IconButton;

public class WindowsTitleBar extends TitleBar {

    private final IconButton closeButton = new IconButton(FluentUiFilledAL.DISMISS_24);

    private final IconButton minimizeButton = new IconButton(FluentUiFilledMZ.MINIMIZE_24);

    private final IconButton maximizeButton = new IconButton(FluentUiFilledMZ.MAXIMIZE_24);

    static {
        val frontendIcon = new FontIcon(FluentUiFilledMZ.MAXIMIZE_16);
        val behind = new FontIcon(FluentUiFilledMZ.MAXIMIZE_16);
    }

    WindowsTitleBar() {
    }
}

File: visual-model-ui/src/main/kotlin/org/visual/model/ui/bar/WindowsTitleBar.kt
package org.visual.model.ui.bar

import org.kordamp.ikonli.fluentui.FluentUiFilledAL
import org.kordamp.ikonli.fluentui.FluentUiFilledMZ
import org.kordamp.ikonli.javafx.FontIcon
import org.visual.model.ui.control.IconButton


class WindowsTitleBar internal constructor() : CommonTitleBar() {
    private val closeButton = IconButton(FluentUiFilledAL.DISMISS_24)

    private val minimizeButton = IconButton(FluentUiFilledMZ.MINIMIZE_24)

    private val maximizeButton = IconButton(FluentUiFilledMZ.MAXIMIZE_24)

    private val frontendIcon = FontIcon(FluentUiFilledMZ.MAXIMIZE_16)
    private val behind = FontIcon(FluentUiFilledMZ.MAXIMIZE_16)


    init {

    }
}
